NeoMathEngine/src/CPU/x86/CpuX86MathEngineBlasMkl.cpp

```diff
@@ -67,18 +63,14 @@ void CCpuMathEngine::multiplyMatrixByMatrix( const CConstFloatHandle& firstHandl
 #endif
 }
 
-void CCpuMathEngine::multiplyMatrixByMatrixAndAdd( const CConstFloatHandle& firstHandle, int firstHeight,
-	int firstWidth, int firstRowSize, const CConstFloatHandle& secondHandle, int secondWidth, int secondRowSize,
-	const CFloatHandle& resultHandle, int resultRowSize, int resultBufferSize )
+void CCpuMathEngine::multiplyMatrixByMatrixAndAdd( const float* first, int firstHeight,
+	int firstWidth, int firstRowSize, const float* second, int secondWidth, int secondRowSize,
+	float* result, int resultRowSize, int resultBufferSize )
 {
 	ASSERT_EXPR( firstWidth <= firstRowSize );
```

Comment:
I know it's questionable, but maybe it's better to remove all the `ASSERT_EXPR` from private methods (for the sake of optimization)?

That will lead to memory errors if the caller has a bug, but it's private and all the callers are other parts of `CCpuMathEngine`...

And leave `ASSERT_EXPR` only in public methods. What do you think?